frontend/src/pages/zaaktypecreate/ZaaktypeCreate.tsx

```diff
@@ -0,0 +1,248 @@
+  const [valuesState, setValuesState] = React.useState<
+    Partial<ZaaktypeCreateLoaderData["results"][0]>
+  >({});
+  const [nonFieldErrors, setNonFieldErrors] = React.useState<
+    string[] | undefined
+  >();
+
+  const [isValidState, setIsValidState] = React.useState(false);
+
+  useEffect(() => {
+    if (actionData?.invalidParams?.length) {
+      setNonFieldErrors(
+        actionData.invalidParams.map(
+          (error) =>
+            `Het veld "${error.name}" is niet correct: ${error.reason}`,
+        ),
+      );
+    } else {
+      setNonFieldErrors(undefined);
+    }
+  }, [actionData]);
+
+  const fields: FormField[] = [
+    {
+      label: "Identificatienummer",
+      name: "identificatie",
+      type: "text",
+      defaultValue: valuesState.waarden?.identificatie || "",
+      required: true,
+    },
+    {
+      label: "Omschrijving",
+      name: "omschrijvingGeneriek",
+      defaultValue: valuesState.waarden?.omschrijvingGeneriek || "",
+      type: "text",
+      required: true,
+    },
+  ];
+
+  const handleSelectTemplate = useCallback(
+    (uuid: string | null) => {
+      setValuesState({});
+      setNonFieldErrors(undefined);
+      if (uuid) {
+        const selectedResult = results.find((result) => result.uuid === uuid);
+        if (selectedResult) {
+          setValuesState({ ...selectedResult });
+        }
+      } else {
+        setValuesState({});
+      }
+    },
+    [results],
+  );
+
+  const handleValidate = useCallback(
+    (values: SerializedFormData, fields: FormField[]) => {
+      const errors = validateForm(values, fields);
+      const isValid = Object.keys(errors).length === 0;
+      setIsValidState(isValid);
+      return errors;
+    },
+    [valuesState],
+  );
+
+  const handleSubmit = useCallback(
+    async (event: FormEvent<HTMLFormElement>, data: SerializedFormData) => {
+      if (!isValidState || !serviceSlug || !catalogusSlug) {
+        console.warn(
+          "Form is not valid or serviceSlug/catalogusSlug is missing.",
+          {
+            isValidState,
+            serviceSlug,
+          },
+        );
+        setNonFieldErrors([
+          "Het formulier is niet correct ingevuld of de service/cataloog is niet ingesteld.",
+        ]);
+        event.preventDefault();
+        return;
+      }
+      const finalPayload: ZaaktypeCreateActionPayload = {
+        zaaktype: {
+          ...valuesState.waarden,
+          ...data,
+        },
+        serviceSlug: serviceSlug,
+        catalogusSlug: catalogusSlug,
+      };
+
+      await submitAction({
+        type: "ZAAKTYPE_CREATE",
+        payload: finalPayload,
+      });
+    },
+    [isValidState],
+  );
+
+  return (
+    <CardBaseTemplate breadcrumbItems={breadcrumbItems}>
+      <Modal open={isFillingForm} onClose={() => setIsFillingForm(false)}>
+        <Body>
+          <H2>Er zijn nog enkele velden nodig om het zaaktype aan te maken</H2>
+          <Form
```

Comment:
I thinkg setting justify to stretch and making the modal (and title) less wide (title shorter...) make this prettier